src/applications/disability-benefits/526EZ/pages/privateMedicalRecordRelease.js

```diff
@@ -0,0 +1,171 @@
+        'ui:options': {
+          itemName: 'Provider',
+          viewField: PrivateProviderTreatmentView,
+          hideTitle: true,
+        },
+        items: {
+          providerFacilityName: {
+            'ui:title': 'Name of private provider or hospital',
+          },
+          'ui:validations': [validateDate],
+          treatmentDateRange: dateRangeUI(
+            'Approximate date of first treatment',
+            'Approximate date of last treatment',
+            'End of treatment must be after start of treatment',
+          ),
+          providerFacilityAddress: merge(addressUI('', false), {
+            street2: {
+              'ui:title': 'Street 2',
+            },
+            postalCode: {
+              'ui:title': 'Postal Code',
+              'ui:options': {
+                widgetClassNames: 'usa-input-medium',
+              },
+              'ui:validations': [
+                {
+                  validator: validateZIP,
+                },
+              ],
+            },
+          }),
+        },
+      },
+    },
+  },
+};
+
+export const schema = {
+  type: 'object',
+  properties: {
+    disabilities: {
+      type: 'array',
+      items: {
+        type: 'object',
+        required: ['providerFacility'],
+        properties: {
+          providerFacility: {
+            type: 'array',
+            minItems: 1,
+            maxItems: 100,
+            items: {
+              type: 'object',
+              required: ['providerFacilityName', 'providerFacilityAddress'],
+              properties: {
+                providerFacilityName: {
+                  type: 'string',
+                  minLength: 1,
+                  maxLength: 100,
+                },
+                treatmentDateRange: {
+                  type: 'object',
+                  properties: {
+                    from: {
+                      $ref: '#/definitions/date',
+                    },
+                    to: {
+                      $ref: '#/definitions/date',
+                    },
+                  },
+                  required: ['from', 'to'],
+                },
+                providerFacilityAddress: {
+                  type: 'object',
+                  required: [
+                    'street',
+                    'city',
+                    'country',
+                    'state',
+                    'postalCode',
+                  ],
+                  properties: {
+                    street: {
+                      minLength: 1,
+                      maxLength: 20,
+                      type: 'string',
+                    },
+                    street2: {
+                      minLength: 1,
+                      maxLength: 5,
+                      type: 'string',
+                    },
+                    city: {
+                      minLength: 1,
+                      maxLength: 30,
+                      type: 'string',
+                    },
+                    postalCode: {
```

Comment:
Are you sure a  e.g. 1-digit or 7 digit postal code would be valid? I thought they could either be 5 or 9 digits but not anything in between?

Reply:
Ahh, I see you're validating this in the `uiSchema`. One thing to think about is that your schema requires only digits be used, but the error message will be 'Please enter a valid 5 or 9 digit ZIP (dashes allowed)', which isn't true in this case. One thing you could do to square the error message with the schema is to use regex and set a pattern on the schema field like what has been done on the mailing address:
```
    zipCode: {
      type: 'string',
      pattern: '^\\d{5}(?:([-\\s]?)\\d{4})?$'
    }
```
In this case, I don't think you'd need the validator.

Reply:
@markgreenburg Agreed, the length needs to go away, and base itself off of the validator. Per requirements they want to keep the error message.